code/Serving_API.py

```diff
@@ -124,9 +146,62 @@ def calculate_shortest_path_distance(G, shortest_path_cities):
 
         total_distance += G[city1][city2]['weight']
         # Only append the coordinates of the source city here
-        coordinate_list.append(G.nodes[city1]['coord'])
-        wkt_list.append(G[city1][city2]['path_wkt'])
-        cable_type_list.append(G[city1][city2]['cable_type'])
+        # for each edge, find all the available asn nodes that are close to the edge
+        insertable_asn = []
```

Comment:
I would not call it asn. ASN stands for AS number. An AS number can correspond to multiple locations. Here you are inserting locations, so I'd just call it insertable_as_locations, or insertable_as_locs, or nearby_as_locs.

Reply:
changed to insertable_as_locs